tests/cupy_tests/manipulation_tests/test_join.py

```diff
@@ -238,12 +239,13 @@ def test_concatenate_dtype_invalid_out(self):
         'unsafe',
     ])
     @testing.for_all_dtypes_combination(names=['dtype1', 'dtype2'])
-    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    @testing.numpy_cupy_array_equal(
+            accept_error=(TypeError, numpy.ComplexWarning))
```

Comment:
Not quite sure that raising is the right path?  But seemed like a decent way of testing the complex warning being raised.